### PR TITLE
ttl: make some integration test faster

### LINKED ISSUE #56718: make ttl unit test faster

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #56718

### What changed and how does it work?

1. Provides `boostJobScheduleForTest` to make some intervals smaller to boost test.
2. Modify some cases to make it more reasonable and faster

In master, run tests in `pkg/ttl/ttlworker` takes 34.5s:

```
> X="run pkg/ttl/ttlworker" make ut
cd tools/check; \
	GO111MODULE=on go build -o ../bin/ut ut.go
cd tools/check; \
	GO111MODULE=on go build -o ../bin/xprog xprog.go
GOBIN=/Users/wangchao/Code/pingcap/tidb/tools/bin GO111MODULE=on go install github.com/pingcap/failpoint/failpoint-ctl@9b3b6e3
tools/bin/ut run pkg/ttl/ttlworker || { find $PWD/ -type d | grep -vE "(\.git|tools)" | xargs tools/bin/failpoint-ctl disable; exit 1; }
2024/10/18 12:09:12 maxprocs: Leaving GOMAXPROCS=10: CPU quota undefined
building task finish, parallelism=20, count=55, takes=1m3.719834292s
run all tasks takes 34.5330225s
```

In this PR, it takes 3.4s:

```
> X="run pkg/ttl/ttlworker" make ut
cd tools/check; \
	GO111MODULE=on go build -o ../bin/ut ut.go
cd tools/check; \
	GO111MODULE=on go build -o ../bin/xprog xprog.go
GOBIN=/Users/wangchao/Code/pingcap/tidb/tools/bin GO111MODULE=on go install github.com/pingcap/failpoint/failpoint-ctl@9b3b6e3
tools/bin/ut run pkg/ttl/ttlworker || { find $PWD/ -type d | grep -vE "(\.git|tools)" | xargs tools/bin/failpoint-ctl disable; exit 1; }
2024/10/18 12:11:16 maxprocs: Leaving GOMAXPROCS=10: CPU quota undefined
building task finish, parallelism=20, count=55, takes=30.296769208s
run all tasks takes 3.442459375s
```

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ttl/ttlworker/job_manager_integration_test.go [hunk 1]
```diff
@@ -416,6 +394,10 @@ func TestTTLJobDisable(t *testing.T) {
 	store, dom := testkit.CreateMockStoreAndDomain(t)
 	tk := testkit.NewTestKit(t, store)
 
+	// turn off the `tidb_ttl_job_enable`
+	tk.MustExec("set global tidb_ttl_job_enable = 'OFF'")
```
Comment (lcwangchao) [inline pkg/ttl/ttlworker/job_manager_integration_test.go:398]: Set the `tidb_ttl_job_enable` to `OFF` before creating TTL table to make test stable

File: pkg/ttl/ttlworker/job_manager_integration_test.go [hunk 2]
```diff
@@ -428,28 +410,18 @@ func TestTTLJobDisable(t *testing.T) {
...
-	tk.MustExec("set global tidb_ttl_job_enable = 'OFF'")
-	defer tk.MustExec("set global tidb_ttl_job_enable = 'ON'")
 
-	retryTime := 15
-	retryInterval := time.Second * 2
-	deleted := false
-	for retryTime >= 0 {
-		retryTime--
-		time.Sleep(retryInterval)
+	time.Sleep(time.Second)
```
Comment (lcwangchao) [inline pkg/ttl/ttlworker/job_manager_integration_test.go:414]: Just sleep 1s to make the test faster. And `boostJobScheduleForTest` can boost the test

File: pkg/ttl/ttlworker/job_manager_integration_test.go [hunk 3]
```diff
@@ -729,39 +700,26 @@ func TestTriggerScanTask(t *testing.T) {
...
 		<-nCli.WatchNotification(context.Background(), "scan")
-		wg.Done()
+		close(done)
 	}()
 	require.NoError(t, m.SubmitJob(se, tblID, tblID, "request1"))
 
 	// notification is sent
-	wg.Wait()
-
-	for time.Now().Before(now.Add(time.Second * 5)) {
```
Comment (lcwangchao) [inline pkg/ttl/ttlworker/job_manager_integration_test.go:751]: What is this case used for? It does not have any assertions. @YangKeao

**Review by YangKeao — APPROVED** [review]
LGTM

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**